### PR TITLE
[BI-677] - PUT traits

### PR DESCRIPTION
**Pull request by ctucker3**
Add PUT traits to the backend. PUT traits accepts a list of traits. Also pushes the changes to the connected brapi server. 

Added a positive test for PUT traits. Needs a couple more negative traits. 

See other PRs:
https://github.com/Breeding-Insight/bi-web/pull/72
https://github.com/Breeding-Insight/sgn/pull/7

### CODE REVIEW

**Review by timparsons — APPROVED** [review]
One small comment that doesn't need to hold up approval.  Looks good!

File: src/main/java/org/breedinginsight/services/TraitService.java
Comment (nickpalladino) [inline src/main/java/org/breedinginsight/services/TraitService.java:377]: We also need to set active right?

File: src/test/java/org/breedinginsight/api/v1/controller/TraitControllerIntegrationTest.java
Comment (ctucker3) [reply src/test/java/org/breedinginsight/api/v1/controller/TraitControllerIntegrationTest.java]: Dang! I would have gotten away with it too if it weren't for you pesky kids! 

I implemented these tests and pushed them up.

File: src/test/java/org/breedinginsight/api/v1/controller/TraitControllerIntegrationTest.java
Comment (ctucker3) [reply src/test/java/org/breedinginsight/api/v1/controller/TraitControllerIntegrationTest.java:965]: Created a new card:

https://breedinginsight.atlassian.net/browse/BI-731

File: src/main/java/org/breedinginsight/services/TraitService.java
Comment (ctucker3) [reply src/main/java/org/breedinginsight/services/TraitService.java:377]: Wouldn't that be in a separate archive endpoint?

File: src/main/java/org/breedinginsight/api/v1/controller/TraitController.java
Comment (ctucker3) [reply src/main/java/org/breedinginsight/api/v1/controller/TraitController.java]: Updated to `Successful Creation`

File: src/main/java/org/breedinginsight/services/TraitService.java
Comment (nickpalladino) [reply src/main/java/org/breedinginsight/services/TraitService.java:377]: based on later discussions, yes

**Review by nickpalladino — APPROVED** [review]

**Pull request merged**
